tests/test_devmgr.cpp

```diff
@@ -39,7 +39,21 @@ class TestDevMgrImp : public DevMgrIface {
 public:
   TestDevMgrImp() {
     // 0 is device_id
-    p_monitor = std::move(PortMonitorIface::make_active(0));
+    // clang++ complains as follows:
+    // ../../tests/test_devmgr.cpp:42:17: error: moving a temporary object prevents
+    //   copy elision [-Werror,-Wpessimizing-move]
+    // p_monitor = std::move(PortMonitorIface::make_active(0));
+    //             ^
+    // ../../tests/test_devmgr.cpp:42:17: note: remove std::move call here
+    // p_monitor = std::move(PortMonitorIface::make_active(0));
+    //             ^~~~~~~~~~                                ~
+	// Disabling the warning seems more appropriate than removing the move
+#pragma GCC diagnostic push
+#if defined(__clang__) && (__clang_major__ > 7)
+#pragma GCC diagnostic ignored "-Wpessimizing-move"
+#endif // __clang__
+	p_monitor = std::move(PortMonitorIface::make_active(0));
```

Comment:
the compiler is right; you should just remove the `move` here. Leaving the `move` prevents the compiler optimization and makes the code less readable.